### PR TITLE
Split afvx to two commands - one for variables and one for function arguments

### LINKED ISSUE #332: Split `afvx` to two commands - one for variables and one for function arguments
**Is your feature request related to a problem? Please describe.**
Currently it's `afvx[j]` returns the output for both local variables and function arguments which is not always desirable.

```
[0x00004d80]> afv?
Usage: afv  [rbs]
| afv*                          output rizin command to add args/locals to flagspace
| afv-([name])                  remove all or given var
| afv=                          list function variables and arguments with disasm refs
| afva                          analyze function arguments/locals
| afvb[?]                       manipulate bp based arguments/locals
| afvd name                     output rizin command for displaying the value of args/locals in the debugger
| afvf                          show BP relative stackframe variables
| afvn [new_name] ([old_name])  rename argument/local
| afvr[?]                       manipulate register based arguments
| afvR [varname]                list addresses where vars are accessed (READ)
| afvs[?]                       manipulate sp based arguments/locals
| afvt [name] [new_type]        change type for given argument/local
| afvW [varname]                list addresses where vars are accessed (WRITE)
| afvx                          show function variable xrefs (same as afvR+afvW)
[0x00004d80]> s main
[0x00004d80]> pd 10
            ; DATA XREF from entry0 @ 0x6b81
╭ 6697: int main (int argc, char **argv);
│ bp: 0 (vars 0, args 0)
│ sp: 17 (vars 17, args 0)
│ rg: 2 (vars 0, args 2)
│           0x00004d80      endbr64
│           0x00004d84      push  r15
│           0x00004d86      push  r14
│           0x00004d88      push  r13
│           0x00004d8a      push  r12
│           0x00004d8c      push  rbp
│           0x00004d8d      push  rbx
│           0x00004d8e      sub   rsp, 0x68
│           0x00004d92      mov   r12, qword [rsi]                     ; argv
│           0x00004d95      mov   rax, qword fs:[0x28]
[0x00004d80]> afvx
afvR
      argv  0x4d92,0x4db0
      argc  0x4dae
   var_58h
   var_40h  0x4fae,0x501c,0x6143,0x685f,0x688d
   var_10h  0x4ffe,0x5cec,0x613b,0x6833,0x6892
   var_1ch  0x53b0,0x53be,0x5ecd
   var_38h  0x5030,0x6091,0x684a,0x6876
    var_8h  0x5279,0x639b,0x6653,0x66a7,0x6933
   var_55h  0x60fb
   var_57h
   var_56h
  var_8h_2
 var_10h_2  0x5aab
   var_28h  0x5ae1,0x5cdb
 var_40h_2  0x5d04
 var_1ch_2
 var_38h_2  0x6aca
 var_55h_2  0x6b44
   var_42h  0x4fd1
afvW
      argv
      argc
   var_58h  0x4d9e
   var_40h  0x608c
   var_10h  0x4fbd,0x5ab9
   var_1ch  0x5028,0x529c
   var_38h  0x505e,0x605a,0x60b7,0x60d1,0x60e2,0x681f,0x687f
    var_8h  0x503c,0x6377,0x6645,0x6929
   var_55h  0x6074,0x60bf
   var_57h  0x607b
   var_56h  0x60da
  var_8h_2  0x5603,0x5915,0x5938,0x5a21,0x5a7a,0x5a8e,0x5b0f
 var_10h_2
   var_28h  0x5ad2,0x5cd1
 var_40h_2
 var_1ch_2  0x6013
 var_38h_2
 var_55h_2  0x6b3d
   var_42h
[0x00004d80]> afvxj
{"reads":[{"name":"argv","addrs":[19858,19888]},{"name":"argc","addrs":[19886]},{"name":"var_58h","addrs":[]},{"name":"var_40h","addrs":[20398,20508,24899,26719,26765]},{"name":"var_10h","addrs":[20478,23788,24891,26675,26770]},{"name":"var_1ch","addrs":[21424,21438,24269]},{"name":"var_38h","addrs":[20528,24721,26698,26742]},{"name":"var_8h","addrs":[21113,25499,26195,26279,26931]},{"name":"var_55h","addrs":[24827]},{"name":"var_57h","addrs":[]},{"name":"var_56h","addrs":[]},{"name":"var_8h_2","addrs":[]},{"name":"var_10h_2","addrs":[23211]},{"name":"var_28h","addrs":[23265,23771]},{"name":"var_40h_2","addrs":[23812]},{"name":"var_1ch_2","addrs":[]},{"name":"var_38h_2","addrs":[27338]},{"name":"var_55h_2","addrs":[27460]},{"name":"var_42h","addrs":[20433]}],"writes":[{"name":"argv","addrs":[]},{"name":"argc","addrs":[]},{"name":"var_58h","addrs":[19870]},{"name":"var_40h","addrs":[24716]},{"name":"var_10h","addrs":[20413,23225]},{"name":"var_1ch","addrs":[20520,21148]},{"name":"var_38h","addrs":[20574,24666,24759,24785,24802,26655,26751]},{"name":"var_8h","addrs":[20540,25463,26181,26921]},{"name":"var_55h","addrs":[24692,24767]},{"name":"var_57h","addrs":[24699]},{"name":"var_56h","addrs":[24794]},{"name":"var_8h_2","addrs":[22019,22805,22840,23073,23162,23182,23311]},{"name":"var_10h_2","addrs":[]},{"name":"var_28h","addrs":[23250,23761]},{"name":"var_40h_2","addrs":[]},{"name":"var_1ch_2","addrs":[24595]},{"name":"var_38h_2","addrs":[]},{"name":"var_55h_2","addrs":[27453]},{"name":"var_42h","addrs":[]}]}
```

**Describe the solution you'd like**
I suggest to split the command in two:
- `afvx` will show both
- `afvxa` will show only arguments
- `afvxv` will show only local variables
- all commands have JSON output.

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This PR splits `afvx` command in two:
- `afvxa` will show only arguments
- `afvxv` will show only local variables
- `afvx` will show both

**Test plan**

Try `afvxa` and `afvxv`

**Closing issues**

Closes https://github.com/rizinorg/rizin/issues/332

cc @no1rr

### CODE REVIEW

**Comment by XVilka** [comment]
@no1rr you need also some tests for this command too.

**Comment by no1rr** [comment]
Test result
```
[0x00401282]> afv?
Usage: afv[?]   # Manipulate arguments/variables in a function
| afv[j*]                  # List all variables and arguments of the current function
| afv=                     # List function variables and arguments with disasm refs
| afv- <varname|*>         # Remove all variables/arguments or just the specified one
| afva                     # Analyze function arguments/locals
| afvd [<varname>]         # Display the value of arguments/variables
| afvf                     # Show BP relative stackframe variables
| afvn <new_name> [<old_name>] # Rename argument/variable in current function
| afvR [<varname>]         # List addresses where vars are accessed (READ)
| afvW [<varname>]         # List addresses where vars are accessed (WRITE)
| afvt <varname> <type>    # Change type for given argument/local
| afvx[jav]                # Show argument/variable xrefs in a function
| afvb[j*-gs]              # Manipulate BP based arguments/locals
| afvr[j*-gs]              # Manipulate register-based arguments/locals
| afvs[j*-gs]              # Manipulate SP based arguments/locals
[0x00401282]> afvx?
Usage: afvx[jav]   # Show argument/variable xrefs in a function
| afvx[j] [<varname>]  # Show function variable xrefs (same as afvR+afvW)
| afvxa[j] [<varname>] # Show function argument xrefs
| afvxv[j] [<varname>] # Show function local variable xrefs
[0x00401282]> afvx
afvR
      arg1  0x40128e
      arg2  0x401292
      arg3  0x401295
   var_18h  0x4012a6,0x4012c0,0x4012d7
   var_1ch  0x401298,0x4012a3,0x4012bd
   var_20h  0x40129b,0x4012a0,0x4012d4
    var_4h  0x4012b7,0x4012ce
afvW
      arg1
      arg2
      arg3
   var_18h  0x40128e
   var_1ch  0x401292
   var_20h  0x401295
    var_4h  0x4012b4
[0x00401282]> afvxa
afvR
      arg1  0x40128e
      arg2  0x401292
      arg3  0x401295
afvW
      arg1
      arg2
      arg3
[0x00401282]> afvxv
afvR
   var_18h  0x4012a6,0x4012c0,0x4012d7
   var_1ch  0x401298,0x4012a3,0x4012bd
   var_20h  0x40129b,0x4012a0,0x4012d4
    var_4h  0x4012b7,0x4012ce
afvW
   var_18h  0x40128e
   var_1ch  0x401292
   var_20h  0x401295
    var_4h  0x4012b4
```

**Comment by wargio** [comment]
you should add this test on the testsuite. if you need help on doing this, let us know.

**Comment by no1rr** [comment]
sorry, I don't know where is the testsuite and how to do with it. Is there any document?

**Comment by wargio** [comment]
You want to have a command test: https://github.com/rizinorg/rizin/blob/dev/test/README.md#commands-tests

**Comment by wargio** [comment]
You can search for tests using as search key `afvx`

**Comment by no1rr** [comment]
add tests for afvxa and afvxv.
https://github.com/rizinorg/rizin/commit/083a65be3b81f84892ac1dae4aaf4c62ed358a40

**Comment by no1rr** [comment]
add tests for afvxa and afvxv
https://github.com/no1rr/rizin/commit/326c6c480c1661281b715755ff70ac9764ea9673

**Comment by XVilka** [comment]
@no1rr thanks! Imported.

**Comment by no1rr** [comment]
My pleasure.

File: librz/core/cmd_analysis.c [hunk 1]
```diff
@@ -899,11 +905,13 @@ static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, c
 	if (name && *name) {
 		var = rz_analysis_function_get_var_byname(fcn, name);
 		if (var) {
-			var_accesses_list(fcn, var, pj, access_type, var->name);
+			if (var->isarg == vlt || vlt == IS_ARG_AND_VAR)
+				var_accesses_list(fcn, var, pj, access_type, var->name);
```
Comment (wargio) [inline librz/core/cmd_analysis.c]:
```suggestion
			if (var->isarg == vlt || vlt == IS_ARG_AND_VAR) {
				var_accesses_list(fcn, var, pj, access_type, var->name);
			}
```

File: librz/core/cmd_analysis.c [hunk 2]
```diff
@@ -899,11 +905,13 @@ static void list_vars(RzCore *core, RzAnalysisFunction *fcn, PJ *pj, int type, c
...
 		}
 	} else {
 		rz_list_foreach (list, iter, var) {
-			var_accesses_list(fcn, var, pj, access_type, var->name);
+			if (var->isarg == vlt || vlt == IS_ARG_AND_VAR)
+				var_accesses_list(fcn, var, pj, access_type, var->name);
```
Comment (wargio) [inline librz/core/cmd_analysis.c]:
```suggestion
			if (var->isarg == vlt || vlt == IS_ARG_AND_VAR) {
				var_accesses_list(fcn, var, pj, access_type, var->name);
			}
```

**Review by wargio — APPROVED** [review]
LGTM

**Pull request merged**